components/brave_wallet/browser/wallet_data_files_installer.cc

```diff
@@ -0,0 +1,170 @@
+
+#include "brave/components/brave_wallet/browser/wallet_data_files_installer.h"
+
+#include <memory>
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/base64.h"
+#include "base/bind.h"
+#include "base/files/file_path.h"
+#include "base/files/file_util.h"
+#include "base/logging.h"
+#include "base/task/thread_pool.h"
+#include "base/values.h"
+#include "brave/components/brave_wallet/common/buildflags/buildflags.h"
+#include "components/component_updater/component_installer.h"
+#include "components/component_updater/component_updater_service.h"
+#include "crypto/sha2.h"
+
+#if BUILDFLAG(BRAVE_WALLET_ENABLED)
+#include "brave/components/brave_wallet/browser/brave_wallet_utils.h"
+#endif
+
+namespace brave_wallet {
+
+namespace {
+
+// CRX public key hash. The extension id is: bbckkcdiepaecefgfnibemejliemjnio
+// Getting the public key:
+// openssl rsa -in ./wallet.pem -pubout -outform DER | openssl base64 -A >
+//   wallet.pub
+// openssl rsa -in ~/Desktop/wallet/wallet.pem -pubout
+//   -outform DER | shasum -a 256 | head -c32 | tr 0-9a-f a-p | mvim -
+const uint8_t kWalletDataFilesSha2Hash[] = {
+    0x11, 0x2a, 0xa2, 0x38, 0x4f, 0x4,  0x24, 0x56, 0x5d, 0x81, 0x4c,
+    0x49, 0xb8, 0x4c, 0x9d, 0x8e, 0xeb, 0xb3, 0xbd, 0x55, 0xdc, 0xf7,
+    0xc0, 0x3e, 0x9b, 0x2a, 0xc2, 0xf5, 0x6a, 0x37, 0x71, 0x67};
+const char kWalletDataFilesDisplayName[] = "Brave Wallet data files";
+const char kWalletBaseDirectory[] = "BraveWallet";
+
+static_assert(base::size(kWalletDataFilesSha2Hash) == crypto::kSHA256Length,
+              "Wrong hash length");
+}  // namespace
+
+class WalletDataFilesInstallerPolicy
+    : public component_updater::ComponentInstallerPolicy {
+ public:
+  WalletDataFilesInstallerPolicy();
+  ~WalletDataFilesInstallerPolicy() override = default;
+
+ private:
+  // The following methods override ComponentInstallerPolicy.
+  bool SupportsGroupPolicyEnabledComponentUpdates() const override;
+  bool RequiresNetworkEncryption() const override;
+  update_client::CrxInstaller::Result OnCustomInstall(
+      const base::DictionaryValue& manifest,
+      const base::FilePath& install_dir) override;
+  void OnCustomUninstall() override;
+  bool VerifyInstallation(const base::DictionaryValue& manifest,
+                          const base::FilePath& install_dir) const override;
+  void ComponentReady(const base::Version& version,
+                      const base::FilePath& path,
+                      std::unique_ptr<base::DictionaryValue> manifest) override;
+  base::FilePath GetRelativeInstallDir() const override;
+  void GetHash(std::vector<uint8_t>* hash) const override;
+  std::string GetName() const override;
+  update_client::InstallerAttributes GetInstallerAttributes() const override;
+
+  void UpdateTokenList(const base::FilePath& install_dir,
+                       std::unique_ptr<base::DictionaryValue> manifest);
+
+  DISALLOW_COPY_AND_ASSIGN(WalletDataFilesInstallerPolicy);
```

Comment:
This is deprecated, please specify these deletion explicitly.

Reply:
Fixed in 00a17f6f17500a72e33cd0760ab2e297b9db06ec